java-package/android/pom.xml

```diff
@@ -0,0 +1,81 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+  <groupId>org.brainflow</groupId>
+  <artifactId>android</artifactId>
+  <version>0.0.1-SNAPSHOT</version>
+  <name>brainflow</name>
+  <description>android binding for brainflow library</description>
+  <url>https://brainflow.org/</url>
+  <packaging>aar</packaging>
+
+  <licenses>
+    <license>
+      <name>MIT</name>
+      <url>https://github.com/brainflow-dev/brainflow/blob/master/LICENSE</url>
+      <distribution>repo</distribution>
+    </license>
+  </licenses>
+
+  <scm>
+    <connection>scm:git:git@github.com:brainflow-dev/brainflow.git</connection>
+    <url>scm:git:git@github.com:brainflow-dev/brainflow.git</url>
+    <developerConnection>scm:git:git@github.com:brainflow-dev/brainflow.git</developerConnection>
+  </scm>
+
+  <developers>
+    <developer>
+      <name>Andrey Parfenov</name>
+      <email>a1994ndrey@gmail.com</email>
+       <organization>brainflow-dev</organization>
+       <organizationUrl>https://github.com/brainflow-dev</organizationUrl>
+    </developer>
+  </developers>
+
+  <distributionManagement>
+    <repository>
+      <id>github</id>
+      <name>GitHub brainflow-dev Apache Maven Packages</name>
+      <url>https://maven.pkg.github.com/brainflow-dev/brainflow</url>
+    </repository>
+  </distributionManagement>
+
+  <build>
+    <finalName>brainflow</finalName>
+
+    <plugins>
+      <plugin>
+        <groupId>com.simpligility.maven.plugins</groupId>
+        <artifactId>android-maven-plugin</artifactId>
+        <version>4.4.3</version>
+        <extensions>true</extensions>
+        <configuration>
+            <ndkArchitecture>arm64-v8a armeabi-v7a x86 x86_64</ndkArchitecture>
+        </configuration>
+      </plugin>
+    </plugins>
+
+  </build>
+  <dependencies>
+    <dependency>
+      <groupId>org.brainflow</groupId>
```

Comment:
it looks a little strange, why do we need brainflow as a dependency inside brainflow pom.xml(I know that its pom.xml for android but still strange)?

Reply:
I'm spending a lot of time looking into options for changing this.  It seems an easier approach to having this not look strange might be to rename the brainflow non-android package to something like 'brainflow-core'.  I am thinking maybe modules but I am very new to maven.  The big issue here is that the maven android plugin does not enforce that transient android dependencies are selected, so if the aar and jar are bundled together in one pom.xml, the wrong jna library is used on android in almost every way of doing it.  profile variables may give access to the entire dependency tree via property reflection, providing a possible avenue for working around this, but it doesn't look very nice and easy.  another approach might be submitting a bugfix to the android maven plugin around the dependency issue, and/or adding a feature to maven to make it easy to depend on parent dependency type; i'm not sure if that would mean requiring users to use a forked plugin version in their android projects or not ... most of the alternatives i'm finding here look kinda large unless we just rename the brainflow dependency or add a comment explaining the situation.  any thoughts?

Reply:
but can it be splitted by 2 different pom.xmls like pom.xml for PC and android-pom.xml for android? This way aar and jar will not be bundled together and may have the same name.

Reply:
And I dont get why in this case you need plain brainflow as a dependency for brainflow android when you can just rebuild it. Looks like I am missing smth here

Reply:
> And I dont get why in this case you need plain brainflow as a dependency for brainflow android when you can just rebuild it. Looks like I am missing smth here

I ran into some [bugs i linked to earlier](https://groups.google.com/g/jna-users/c/QUFT16yJyt0/m/gapq1WL9AQAJ) between `maven-compiler-plugin` and `maven-android-plugin`.  After your replies, I spent some more time with this and found ways of handling them (I was wrong; there is actually a new release where some of the issues are fixed; maven-compiler-plugin 3.8.1).  So, this is looking somewhat more possible.

> but can it be splitted by 2 different pom.xmls like pom.xml for PC and android-pom.xml for android? This way aar and jar will not be bundled together and may have the same name.

If they have the same name but are separate pom files, then if they are both installed to the same (even local) maven repository, the later-installed pom.xml overwrites the earlier-installed pom.xml .  For this situation, it might make more sense to either give them separate names, or merge the pom files into one (which may be doable without profile tricks .. might look at that more now that sources can be built under android).

Reply:
we can change name for aar package, its ok